### DISCUSSION TITLE
Are low-level Jets feasible?

### DISCUSSION THREAD

**Discussion by hsjoberg** [BlockstreamResearch/simplicity#91] [Q&A] [answered]
Greetings.
I'm not very knowledgeable about these things. This might be a stupid question, so bear with me.

It is my understanding that the concept of Jets exists to off-load the burden of having to execute heavy computations in pure Simplicity. Thus, making Jets out of common Bitcoin-related use-cases like Schnorr signatures and SHA256 hashing makes sense, because it would otherwise be too expensive to accomplish these things.

My question is if it's also feasible to make Jets of low-level constructs, like 32-bit arithmetic, math operations and whatnot?
The rationale behind this is that it could make more experimental things plausible, albeit perhaps still a bit costly. A hypothetical example would be BLS signatures.
If something would gain traction, they could then be made into high-level Jets.

**Comment by apoelstra** [comment, marked as answer]
Yep! In fact this is the plan. @roconnor-blockstream I think can say more about what the set of jets will look like for our initial release.

**Reply by roconnor-blockstream** [reply]
Appendix B (at the time of writing) in [the Simplicity Tech Report](https://github.com/ElementsProject/simplicity/blob/pdf/Simplicity-TR.pdf) lists the jets I'm aiming to have for our initial release.  It just a proposal though and it is subject to change.

**Reply by hsjoberg** [reply]
Thank you for your answers. 💯
